File: py/IEC61970/Base/Core/Substation.py
# from equipment_container import EquipmentContainer
# from voltage_level import VoltageLevel
# from bay import Bay
# from dc_converter_unit import DCConverterUnit
# from feeder import Feeder
from CIM_STD_PYTHON.TC57CIM.IEC61970.Base.Core.Bay import Bay
from CIM_STD_PYTHON.TC57CIM.IEC61970.Base.Core.EquipmentContainer import EquipmentContainer
from CIM_STD_PYTHON.TC57CIM.IEC61970.Base.Core.Feeder import Feeder
from CIM_STD_PYTHON.TC57CIM.IEC61970.Base.Core.VoltageLevel import VoltageLevel
from CIM_STD_PYTHON.TC57CIM.IEC61970.Base.DC.DCConverterUnit import DCConverterUnit


class Substation(EquipmentContainer):
    def __init__(self):
        super().__init__()
        self.voltage_levels = VoltageLevel()  # The voltage levels within this substation.
        self.bays = [Bay()]  # Bays contained in the substation.
        self.dc_converter_unit = DCConverterUnit()  # The DC converter unit belonging to the substation.
        self.normal_energized_feeder = Feeder()  # The normal energized feeders of the substation. Also used for naming purposes.

File: java/IEC61970/Base/Core/Substation.java
package IEC61970.Base.Core;

import IEC61970.Base.DC.DCConverterUnit;

/**
 * A collection of equipment for purposes other than generation or utilization,
 * through which electric energy in bulk is passed for the purposes of switching
 * or modifying its characteristics.
 * @created 02-Jan-2024 10:37:56 PM
 */
public class Substation extends EquipmentContainer {

	/**
	 * The voltage levels within this substation.
	 */
	public VoltageLevel VoltageLevels;
	/**
	 * Bays contained in the substation.
	 */
	public Bay Bays;
	/**
	 * The DC converter unit belonging of the substation.
	 */
	public DCConverterUnit DCConverterUnit;
	/**
	 * The normal energized feeders of the substation. Also used for naming purposes.
	 */
	public Feeder NormalEnergizedFeeder;

	public Substation(){

	}
}//end Substation